src/qt/breakpoint_window.cpp

```diff
@@ -0,0 +1,504 @@
+
+    // check to see if it overflows:
+    T r = (T)v;
+
+    if(r != v)
+    {
+        QMessageBox messageBox;
+        messageBox.critical(0,"Error","Overflow of breakpoint value");
+    }
+
+    return r;
+}
+
+template uint8_t breakpoint_window::get_value();
+template uint16_t breakpoint_window::get_value();
+template uint32_t breakpoint_window::get_value();
+template uint64_t breakpoint_window::get_value();
+
+DebuggerBreakpoint* breakpoint_window::make_memory_breakpoint(uint32_t addr, bool limit_to_32)
+{
+    char buff[64];
+    sprintf(buff, "0x%x", addr);
+    address_line->setText(QString(buff));
+    return make_memory_breakpoint(limit_to_32);
+}
+
+DebuggerBreakpoint *breakpoint_window::make_register_breakpoint(std::vector<EERegisterDefinition>* available_regs)
+{
+    ee_regs = available_regs;
+    setWindowTitle("Register Breakpoint");
+    address_line->hide();
+    address_label->hide();
+    setup_comparison_combo_boxes();
+
+    // set up the register combos
+    for(auto name : ee_register_kind_names)
+        reg_type_combo->addItem(QString(name));
+
+    on_reg_type_combo_currentIndexChanged(0);
+    this->exec();
+
+    if(this->result() == QDialog::Rejected || !register_selected)
+        return nullptr;
+
+    auto data_type = (EmotionBreakpoints::MemoryComparisonType)data_type_combo->currentIndex();
+    auto data_kind = (EmotionBreakpoints::MemoryComparisonKind)comparison_combo->currentIndex();
+    bool clear = clear_checkbox->isChecked();
+    auto reg_def = (*available_regs)[combo_reg_idx[reg_combo->currentIndex()]];
+
+    return make_register_breakpoint(reg_def, data_type, data_kind, clear);
+}
+
+DebuggerBreakpoint *breakpoint_window::make_register_breakpoint(std::vector<IOPRegisterDefinition>* available_regs)
+{
+    iop_regs = available_regs;
+    setWindowTitle("Register Breakpoint");
+    address_line->hide();
+    address_label->hide();
+    setup_comparison_combo_boxes(true);
+
+    // set up the register combos
+    for(auto name : iop_register_kind_names)
+        reg_type_combo->addItem(QString(name));
+
+    on_reg_type_combo_currentIndexChanged(0);
+    this->exec();
+
+    if(this->result() == QDialog::Rejected || !register_selected)
+        return nullptr;
+
+    auto data_type = (EmotionBreakpoints::MemoryComparisonType)data_type_combo->currentIndex();
+    auto data_kind = (EmotionBreakpoints::MemoryComparisonKind)comparison_combo->currentIndex();
+    bool clear = clear_checkbox->isChecked();
+    auto reg_def = (*available_regs)[combo_reg_idx[reg_combo->currentIndex()]];
+
+    return make_register_breakpoint(reg_def, data_type, data_kind, clear);
+}
+
+DebuggerBreakpoint *breakpoint_window::make_register_breakpoint(EERegisterDefinition &def)
+{
+    setWindowTitle("Register Breakpoint");
+    address_line->hide();
+    address_label->hide();
+    reg_combo->hide();
+    reg_type_combo->hide();
+    reg_type_label->hide();
+    reg_label->setText("Register: " + QString(def.get_name().c_str()));
```

Comment:
Can use QString::arg here instead of using concatenation.

```suggestion
    reg_label->setText(QString("Register: %1").arg(def.get_name().c_str()));
```